### PR TITLE
docs: Sandbox cleanups/updates

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message:  docs: Sandbox cleanups/updates
Additional Description:

first of several cleanup/update PRs for sandboxes.

Adds a separate page about sandbox setup/deps

and for each sandbox this will:

- remove instructions about setup
- add a requirements box pointing to setup and any additional software used in the documented example
- add a see-also box pointing to other relevant sandboxes/pages
- minor cleanups as noticed
- improve linking and other x-referencing

Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

**Comment by phlax** [comment]
apologies for api/deps ping it was a merge/rebase conflict

**Comment by phlax** [comment]
this is still pretty wip, but you can get an idea of the direction from current render:

- https://storage.googleapis.com/envoy-pr/1d59439/docs/start/sandboxes/index.html
- https://storage.googleapis.com/envoy-pr/1d59439/docs/start/sandboxes/cache.html

File: docs/root/start/sandboxes/setup.rst [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+.. _start_sandboxes_setup_docker_compose:
+
+Install Docker Compose
+----------------------
+
+You will need a fairly recent version of docker-compose.
+
+Most of the examples use version ``3.7`` of the docker-compose configuration, so your version
+should be able to parse this configuration.
```
Comment (phlax) [inline docs/root/start/sandboxes/setup.rst:25]: get some actual versions

**Comment by phlax** [comment]
This is the first pass at cleaning up the sandboxes.

With the next PR i would like to do some cleanups to the `examples/` side of the sandboxes (rather than docs)

Im pretty happy with the outcome so far, so i reckon its ready for final review.

Last render can be seen here:

https://storage.googleapis.com/envoy-pr/ba2b126/docs/start/sandboxes/index.html

**Review by mattklein123 — APPROVED** [review]
Awesome!

**Pull request merged**